salt/netapi/rest_cherrypy/app.py

```diff
@@ -1427,6 +1448,14 @@ def POST(self, **kwargs):
         else:
             creds = cherrypy.serving.request.lowstate
 
+        # Authorization header
+        remote_user = cherrypy.request.headers.get('X-Remote-User')
+        sharedsecret =  cherrypy.request.headers.get('X-Shared-Secret')
+        if remote_user and sharedsecret:
+            creds['username'] = username
+            creds['sharedsecret'] = sharedsecret
+            creds['eauth'] = 'sharedsecret'
+
```

Comment:
Ah, you're right that is a bad hack. I _think_ you can add arbitrary data to an Authorization header but I'm not sure, the spec is a little dense reading. :smile: If not then I'm comfortable with your original suggestion of custom header, e.g., `X-Authorization-Eauth` or something.